### PR TITLE
feat: cache daemon client for each pipeline

### PR DESCRIPTION
**Pull request by dpadhiar**
Explain what this PR does.

Cache client for each pipeline to reduce creation of them for each call to pipeline APIs.

### CODE REVIEW

**Comment by whynowy** [comment]
Refer to https://github.com/numaproj/numaflow/blob/main/pkg/reconciler/vertex/scaling/scaling.go#L51

File: server/apis/v1/handler.go [hunk 1]
```diff
@@ -264,13 +270,16 @@ func (h *handler) GetPipeline(c *gin.Context) {
 	}
 
 	// get pipeline lag
-	// TODO: the client should be cached.
-	client, err := daemonclient.NewDaemonServiceClient(daemonSvcAddress(ns, pipeline))
-	if err != nil {
-		h.respondWithError(c, fmt.Sprintf("Failed to fetch pipeline: failed to calculate lag for pipeline %q: %s", pipeline, err.Error()))
-		return
+	dClient, _ := h.daemonClientsCache.Get(daemonSvcAddress(ns, pipeline))
```
Comment (whynowy) [inline server/apis/v1/handler.go:273]: make it a function?

File: server/apis/v1/handler.go [hunk 2]
```diff
@@ -313,13 +319,11 @@ func (h *handler) GetPipeline(c *gin.Context) {
 	}
 
 	// get pipeline lag
-	// TODO: the client should be cached.
-	client, err := daemonclient.NewDaemonServiceClient(daemonSvcAddress(ns, pipeline))
-	if err != nil {
-		h.respondWithError(c, fmt.Sprintf("Failed to fetch pipeline: failed to calculate lag for pipeline %q: %s", pipeline, err.Error()))
+	client, err := h.getDaemonClient(ns, pipeline)
+	if err != nil || client == nil {
+		h.respondWithError(c, fmt.Sprintf("failed to get daemon service client for pipeline %q, %s", pipeline, err.Error()))
```
Comment (jy4096) [inline server/apis/v1/handler.go:324]: use gin context to set the response? ...

Comment (jy4096) [reply server/apis/v1/handler.go:324]: where is `respondWithError` defined...

Comment (jy4096) [reply server/apis/v1/handler.go:324]: nvm found it...

**Review by whynowy — APPROVED** [review]

**Pull request merged**